src/coreclr/jit/typelist.h

```diff
@@ -62,6 +62,7 @@ DEF_TP(SIMD8    ,"simd8"  , TYP_SIMD8,   TI_STRUCT, 8, 8,  8,   2, 8, VTF_S)
 DEF_TP(SIMD12   ,"simd12" , TYP_SIMD12,  TI_STRUCT,12,16, 16,   4,16, VTF_S)
 DEF_TP(SIMD16   ,"simd16" , TYP_SIMD16,  TI_STRUCT,16,16, 16,   4,16, VTF_S)
 DEF_TP(SIMD32   ,"simd32" , TYP_SIMD32,  TI_STRUCT,32,32, 32,   8,16, VTF_S)
+DEF_TP(SIMD64   ,"simd64" , TYP_SIMD64,  TI_STRUCT,64,64, 64,   16,16, VTF_S)
```

Comment:
It's also a little surprising to me that SIMD32 alignment is 16, not 32, and SIMD64 is also 16, not 64, but the comment above indicates there's a reason for that (maybe it should point at code that changes the alignment dynamically).

Reply:
We don’t actually use it to my knowledge and any handling should be superfluous 

it likely just was never ifdefd out when Arm64 was brought online

Reply:
I do not fully understand the question here
1. Do you mean the types `SIMD32` and `SIMD64` should not be defined for Arm64 at all? The following made sense in my head since both arm and x86 use the same calls to create `GenTreeVecCon` etc and if `GenTreeVecCon` is using `simd64Val` now since it's the largest, we need to do the same here : 
https://github.com/dotnet/runtime/pull/80960/files#diff-547b75c8f8dd006693805c30d7a0e2ddcdd6b10691383dfbb16e3ed0c50f236aL1431-R1431 . When you say it doesn't need to be handled, do you mean the changes inside `Lowering::LowerHWIntrinsicCreate` above are not needed?
2. Not sure about the alignment being 16 there. That was another of my opens

Reply:
Just saying that `TYP_SIMD32/64` paths aren't ever used on arm64 and so we could ifdef them out and only have them on xarch.

I don't remember all the things that the `alignment: 16` impacts. I believe it primarily impacts stack alignment and I don't believe we have the necessary support to change it to 32/64 today.

Reply:
I think @DeepakRajendrakumaran doesn't need to worry about the impact of TYP_SIMD64 on Arm64 *too much*, but we'll see what the throughput (TP) measurements say about the actual impact. Perhaps we do need to go back and make sure Arm64 isn't unduly impacted in TP or JIT code size by unused functionality.

One example: a lot of this PR is adjusting simd32_t to simd64_t in many places. Maybe there should be a "union" type of simd_t (or simd_data_t?) that is a union of simd64_t, simd32_t, simd16_t, etc. Except on Arm64 it wouldn't include simd32_t or simd64_t, since they aren't needed/used.

Reply:
> One example: a lot of this PR is adjusting simd32_t to simd64_t in many places. Maybe there should be a "union" type of simd_t (or simd_data_t?) that is a union of simd64_t, simd32_t, simd16_t, etc. Except on Arm64 it wouldn't include simd32_t or simd64_t, since they aren't needed/used.

Each type is functionally that already.

That is, `simd16_t` is a union of each primitive type + 2x simd8_t
`simd32_t` is likewise of each primitive type + 4x simd8_t and 2x simd16_t

So by using `simd64_t` you have efficient access to just the `simd8/16/32` data if necessary. The downside being that it declares 48 bytes of unused data on Arm64, and 32 bytes of unused data on hardware without AVX-512 support.